dascore/core/inventory.py

```
@@ -988,17 +1066,16 @@ def coordinates_at(self, distances) -> np.ndarray:
-            seg_coords = segment.interpolate(dist)
-            filled = ~np.isnan(seg_coords[:, 0])
-            out[filled] = seg_coords[filled]
-        # The outermost endpoint of the geometry coverage domain is included.
-        outer = max(seg.interval[1] for seg in self.geometry)
-        at_outer = dist == outer
-        if np.any(at_outer):
-            segment = max(self.geometry, key=lambda s: s.interval[1])
-            coords = np.asarray(segment.coordinates, dtype=float)
-            out[at_outer] = coords[-1]
+        masks = interval_masks(dist, [x.interval for x in self.geometry])
+        for segment, mask in zip(self.geometry, masks):
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)